geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java

```diff
@@ -846,7 +846,14 @@ private boolean setServerNames(SSLParameters modifiedParams, HostAndPort addr) {
         // ignore - we'll see what happens with endpoint validation using a numeric address...
       }
     }
-    serverNames.add(new SNIHostName(hostName));
+    try {
+      serverNames.add(new SNIHostName(hostName));
+    } catch (Exception ex) {
```

Comment:
If we can't establish the SNI field as requested shouldn't we throw an informative exception?   If you throw an SSLException, for instance, ConnectionFactoryImpl will catch it and log it.

Reply:
Hi @bschuchardt ,
What do you mean by throwing an informative exception?
Is it ok to log it in warn level?

Reply:
throw new SSLException("unable to establish requested server name", ex)

Reply:
I'm not sure that's actually necessary though - ConnectionFactoryImpl already catches the exception this PR is guarding and logs it.  Do you have a failure that this PR is fixing?

Reply:
This issue didn't exist before adding [GEODE-7852](https://issues.apache.org/jira/browse/GEODE-7852).
I can't find it where it catches this exception from the [ticket description](https://issues.apache.org/jira/browse/GEODE-8655) and where it logs it. This catch is added to handle this exception:
` java.lang.IllegalArgumentException: Contains non-LDH ASCII characters`
I described a failure in the ticket, and need a help to write test for it as it is using ipv6 and TLS. 
You can try to reproduce it locally. If you need more information to reproduce the issue please contact me via mail or slack.

Reply:
I'm removing myself as a reviewer of this PR.  I will not be working on Apache Geode in the near future and should not be blocking any PRs.